cf-oscore/src/main/java/org/eclipse/californium/cose/EncryptCommon.java

```diff
@@ -218,4 +232,20 @@ private void AES_CCM_Encrypt(AlgorithmID alg, byte[] rgbKey) throws CoseExceptio
 	public void setEncryptedContent(byte[] rgb) {
 		rgbEncrypt = rgb;
 	}
+
+	/**
+	 * Get IV length in bytes.
+	 */
+	public static int ivLength(AlgorithmID alg) {
+		switch (alg) {
+		case AES_CCM_16_64_128:
+		case AES_CCM_16_128_128:
+			return AES_CCM_16_IV_LENGTH;
+		case AES_CCM_64_64_128:
+		case AES_CCM_64_128_128:
+			return AES_CCM_64_IV_LENGTH;
+		default:
+			return -1;
+		}
+	}
```

Comment:
Maybe adding a  `boolean isSupportedAesCcm(AlgorithmID alg)` obsoletes some general `switch(alg)` as above in line 69 or 91. Or use return value of this `ivLength()` instead of the `switch(alg)`.